ir/expression.cpp

```diff
@@ -90,17 +90,32 @@ void IR::Constant::handleOverflow(bool noWarning) {
     big_int mask = Util::mask(width);
 
     if (tb->isSigned) {
+        auto masked_value = value;
```

Comment:
The logic here looks fairly complex. Possible to add some tests to https://github.com/p4lang/p4c/blob/main/test/gtest/constant_expr_test.cpp which exercise this code?